tools/cli/domainCommands.go

```diff
@@ -415,31 +417,50 @@ VisibilityArchivalURI: {{.}}{{end}}
 {{with .FailoverInfo}}Graceful failover info:
 {{table .}}{{end}}`
 
-var newtemplateDomain = `Validation Check: 
+var newtemplateDomain = `Validation Check:
 {{- range .}}
 - {{.ValidationCheck}}: {{.ValidationResult}}
-  {{- with .ValidationDetails}}
-    {{- with .CurrentDomainRow}}
-      Current Domain:
-        Name: {{.DomainInfo.Name}}
-        UUID: {{.DomainInfo.UUID}}
-    {{- end}}
-    {{- with .NewDomainRow}}
-      New Domain:
-        Name: {{.DomainInfo.Name}}
-        UUID: {{.DomainInfo.UUID}}
-    {{- end}}
-    {{- if .LongRunningWorkFlowNum}}
-      Long Running Workflow Num: {{.LongRunningWorkFlowNum}}
-    {{- end}}
-    {{- range .MismatchedDynamicConfig}}
-      Mismatched Dynamic Config:
-        Current Response: 
-          Value: {{.CurrResp.Value}}
-        New Response: 
-          Value: {{.NewResp.Value}}
-    {{- end}}
-  {{- end}}
+{{- with .ValidationDetails}}
+ {{- with .CurrentDomainRow}}
+ Current Domain:
+   Name: {{.DomainInfo.Name}}
+   UUID: {{.DomainInfo.UUID}}
+ {{- end}}
+ {{- with .NewDomainRow}}
+ New Domain:
+   Name: {{.DomainInfo.Name}}
+   UUID: {{.DomainInfo.UUID}}
+ {{- end}}
+ {{- if .LongRunningWorkFlowNum}}
+ Long Running Workflow Num: {{.LongRunningWorkFlowNum}}
+ {{- end}}
+ {{- range .MismatchedDynamicConfig}}
+ Mismatched Dynamic Config:
+ Config Key: {{.Key}}
+   {{- range $index, $value := .CurrValues}}
+   {{- if eq $index 0 }}
+   Current Response:
+   {{- end}}
+     Data: {{ printf "%s" $value.Value.Data }}
+     Filters:
+     {{- range $filter := $value.Filters}}
+       - Name: {{ $filter.Name }}
+         Value: {{ printf "%s" $filter.Value }}
+     {{- end}}
+   {{- end}}
+   {{- range $index, $value := .NewValues}}
+   {{- if eq $index 0 }}
+   New Response:
+   {{- end}}
+     Data: {{ printf "%s" $value.Value.Data }}
+     Filters:
+     {{- range $filter := $value.Filters}}
+       - Name: {{ $filter.Name }}
+         Value: {{ printf "%s" $filter.Value }}
+     {{- end}}
+   {{- end}}
+ {{- end}}
```

Comment:
template is slightly off. below works after testing
```suggestion
 {{- range .MismatchedDynamicConfig}}
 {{ $dynamicConfig := . }}
 Mismatched Dynamic Config:
 Config Key: {{.Key}}
   {{- range $i, $v := .CurrValues}}
   Current Response:
     Data: {{ printf "%s" (index $dynamicConfig.CurrValues $i).Value.Data }}
     Filters:
     {{- range $filter := (index $dynamicConfig.CurrValues $i).Filters}}
       - Name: {{ $filter.Name }}
         Value: {{ printf "%s" $filter.Value.Data }}
     {{- end}}
   New Response:
     Data: {{ printf "%s" (index $dynamicConfig.NewValues $i).Value.Data }}
     Filters:
     {{- range $filter := (index $dynamicConfig.NewValues $i).Filters}}
       - Name: {{ $filter.Name }}
         Value: {{ printf "%s" $filter.Value.Data }}
     {{- end}}
   {{- end}}
 {{- end}}
```